### DISCUSSION TITLE
Svelte: force-reloading a query

### DISCUSSION THREAD

**Discussion by trevyn** [urql-graphql/urql#1353] [Q&A] [answered]
Right now I have something like this:

```svelte
<Button
 on:click={async () => {
  await shuffleCardsMutation();
  $listCardsQuery.context = { requestPolicy: 'cache-and-network', forceUpdate: Date.now() };
 }}>
 Shuffle Cards
</Button>
```

This works, but setting the `context` to a cache-buster feels really old-school and awkward, because it's exposing an implementation detail.

Is this currently the most idiomatic way of getting this result?

**Comment by kitten** [comment, marked as answer]
You actually shouldn't need to set `forceUpdate`. Instead the most idiomatic method here is to set `requestPolicy` to `networkOnly`: https://formidable.com/open-source/urql/docs/concepts/core-package/

The problem here is partly reactivity in Svelte being a little odd compared to other frameworks as it doesn't appear functional, but all that's being done here is updating the inputs to the query call which will automatically restart it
